modules/ROOT/pages/about-classloading-isolation.adoc

```diff
@@ -112,14 +112,22 @@ Application Code::
 
 == Shared Libraries
 
-All JAR (`.jar`) dependencies declared in the application's `pom.xml` file are visible to the application's classloader but not visible to the classloader of each connector used in the application. Because the classloading mechanism isolates each connector to prevent it from accessing classes from other connectors, you need to declare connector dependencies as shared libraries.
+All dependencies (`.jar` files for example) declared in the application's `pom.xml` file are visible to the application's classloader but not visible to the classloader of each connector used in the application. The classloading mechanism isolates each connector to prevent it from accessing classes from other connectors. If a connector's classloader needs access to an application dependency, you need to declare this dependency as a shared library.
 
-Consider an application that uses the Java connector and the connector needs to use a class that is part of a JAR dependency declared in the application's `pom.xml` file, this is not possible because the connector's classloader is not able to find the class. To make this class visible to the connector, you must declare the dependency that contains the class as a shared library in the Mule Maven Plugin configuration of your application's `pom.xml` file.
+Consider an application that uses the Java connector and the connector needs to use a class that is part of a JAR dependency declared in the application's `pom.xml` file, this is not possible because the connector's classloader is not able to find that class. To make this class visible to the connector, you must declare the dependency that contains the class as a shared library in the Mule Maven Plugin configuration of your application's `pom.xml` file.
 
 If you use Anypoint Studio or Flow Designer to configure a connector that uses external libraries, the dependencies are automatically added as shared libraries. For example, if you add a Database Connector to your application, and then configure the connection driver using Anypoint Studio, the driver is automatically added as a shared library in your project's `pom.xml` file.
 
 See xref:mmp-concept.adoc#configure-shared-libraries[Configure Shared Libraries] for configuration instructions.
 
+=== Considerations
+
+Shared libraries are visible to all connectors in the application. Configuring shared libraries can cause issues if the shared library has different versions of the same classes that exist in other connectors or their dependencies.
+
+In case you need to configure a different version of the same dependency for different connectors, configure them as plugin dependencies. Plugin dependencies are configured at connector level and are visible only to that connector.
```

Comment:
I would change it to something like: 
For those cases, there is another option you can use to enrich a connector's classpath from an application: Additional Plugin Dependencies. 
With this approach, instead of sharing an application dependency, you can effectively add new dependencies to a connector as if they were declared in its pom.xml. 
That means, not only that those new dependencies will exclusively be available to that connector, but that the application does not have the need to declare them as it own. (Also, the dependencies version must be specified since they will no longer be available from the application dependencies declaration.)